### DISCUSSION TITLE
[MongoDB] Using Prisma's Field References to Compare a Field to a Calculated Value of another Field

### DISCUSSION THREAD

**Discussion by bn-piar** [prisma/prisma#18033] [Q&A] [answered]
I have a model 
```prisma

model Product {
  total Int
  available Int
}
```
I need to find all documents whose `available` is some percent of total, for example whose `available` is 80% of total, something like this

```ts
const products = await db.product.findMany({
  where: {
    available: { gte: db.product.fields.total * 0.8 } // not working
  }
});
```
Is there a way to evaluate `db.product.fields.total` to a value and then execute search without using raw queries?

**Comment by ludralph** [comment, marked as answer]
Hi @bn-piar 👋 

You can't access the `total` field as defined in your code. This is an excellent case for using the [client extension](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions). Could you take a look and let me know if you have any other questions?

**Reply by bn-piar** [reply]
@ludralph hey, thanks for the answer. Are you suggesting to use custom queries or results? Could you provide a simple example please?

**Reply by ludralph** [reply]
Hi @bn-piar 
You could do something like this 

```js
const xprisma = prisma.$extends({
    result: {
      product: {
        save: {
          needs: { total: true, available: true },
          compute(product) {
            return () =>
              prisma.product.findFirst({ where: { available: { gte: product.total * 0.8 }}})
          },
        },
      },
    },
    
  })

const res = await xprisma.product.findFirst({ take: 1});
 console.log('PPP ', await  res.save())

This gives an output 

prisma:query db.Product.aggregate([ { $sort: { _id: 1, }, }, { $limit: 1, }, { $project: { _id: 1, total: 1, available: 1, }, }, ])
prisma:query db.Product.aggregate([ { $match: { $expr: { $and: [ { $gte: [ "$available", { $literal: 400, }, ], }, { $ne: [ "$available", "$$REMOVE", ], }, ], }, }, }, { $limit: 1, }, { $project: { _id: 1, total: 1, available: 1, }, }, ])
PPP  { id: '63f71dbc338ef58b586de9c3', total: 1000, available: 800 }
```
You can adapt this to your use case.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
